recipes/qt/5.x.x/conanfile.py

```diff
@@ -666,8 +666,8 @@ def build(self):
         for package in self.deps_cpp_info.deps:
             args += ["-I \"%s\"" % s for s in self.deps_cpp_info[package].include_paths]
             args += ["-D %s" % s for s in self.deps_cpp_info[package].defines]
-        lib_arg = "/LIBPATH:" if self._is_msvc else "-L"
-        args.append("QMAKE_LFLAGS+=\"%s\"" % " ".join("%s%s" % (lib_arg, l) for package in self.deps_cpp_info.deps for l in self.deps_cpp_info[package].lib_paths))
+        args.append("QMAKE_LIBDIR+=\"%s\"" % " ".join("%s" % l for package in self.deps_cpp_info.deps for l in self.deps_cpp_info[package].lib_paths))
+        args.append("QMAKE_RPATHDIR+=\"%s\"" % " ".join("%s" % l for package in self.deps_cpp_info.deps for l in self.deps_cpp_info[package].lib_paths))
```

Comment:
```suggestion
        args.append("QMAKE_RPATHLINKDIR+=\"%s\"" % " ".join(l for package in self.deps_cpp_info.deps for l in self.deps_cpp_info[package].lib_paths))
```
`QMAKE_RPATHDIR` hardcodes some paths to help find libraries at run time. This is not a good thing, cf https://github.com/conan-io/conan-center-index/blob/154c9edca773d52a790d917bf2c0b95796a7ce7b/docs/packaging_policy.md#package.
I think what we need is to help LD find the libraries at link-time, which is what `QMAKE_RPATHLINKDIR` is made for

Reply:
I tried `QMAKE_RPATHLINKDIR` first actually, but this resulted in an error from `ld`. I'm rebuilding now so that I can get that error message.

Reply:
Here's the error:

```
/usr/bin/ld: error: /var/home/jordan/.conan/data/pcre2/10.39/_/_/package/61657e94fb41f62f4a54e6bbefdc6cfcf2dc949c/lib: read: Is a directory
```

And the full log: [conan-create-qt-5-qtwebengine-rpath-link-dir.txt](https://github.com/conan-io/conan-center-index/files/9047050/conan-create-qt-5-qtwebengine-rpath-link-dir.txt)

Reply:
Mmm, maybe the makefile would help to understand this error. It's not clear from the error message

Reply:
I figured it out. I had to pass the directories to `QMAKE_RPATHLINKDIR` as a colon-separated list instead of a space-separated list. Works!

Reply:
Nice catch again.
I guess we will soon see if it works for windows too, or if we need to use `os.pathsep` instead
It's surprising how it works with a space separator with `QMAKE_LIBDIR`